internal/core/src/query/visitors/ExecExprVisitor.cpp

```diff
@@ -501,6 +501,63 @@ ExecExprVisitor::ExecUnaryRangeVisitorDispatcher(UnaryRangeExpr& expr_raw)
     return ExecUnaryRangeVisitorDispatcherImpl<T>(expr_raw);
 }
 
+template <typename T>
+bool
+CompareTwoJsonArray(T arr1, const proto::plan::Array& arr2) {
+    int json_array_length = 0;
+    if constexpr (std::is_same_v<
+                      T,
+                      simdjson::simdjson_result<simdjson::ondemand::array>>) {
+        json_array_length = arr1.count_elements();
+    }
+    if constexpr (std::is_same_v<T,
+                                 std::vector<simdjson::simdjson_result<
+                                     simdjson::ondemand::value>>>) {
+        json_array_length = arr1.size();
+    }
+    if (arr2.array_size() != json_array_length) {
+        return false;
+    }
+    int i = 0;
+    for (auto&& it : arr1) {
+        switch (arr2.array(i).val_case()) {
```

Comment:
same problem as up, switch out of for is better for performance

Reply:
This is different from the above. The elements here may be different. For example, data is: `{"x": [1, 2.2, false, "abc"]}`, the expression is `x == [1, 2.2, false, " abc"]`.